### PR TITLE
Shuffle and repeat

### PR DESCRIPTION
**Pull request by incompl**

### CODE REVIEW

File: src/playlist-maker.js [hunk 1]
```diff
@@ -120,6 +128,11 @@ const factory = (player, initialList, initialIndex = 0) => {
   const playlist = player.playlist = function(newList, newIndex = 0) {
     if (Array.isArray(newList)) {
       list = newList.slice();
+
+      let first = list.shift();
+      shuffle(list);
```
Comment (gkatsev) [inline src/playlist-maker.js]: Does this mean the playlist always shuffles when set?

File: src/playlist-maker.js [hunk 2]
```diff
@@ -248,11 +263,25 @@ const factory = (player, initialList, initialIndex = 0) => {
...
-        return list[playlist.currentItem(index)];
+      // Default behavior
+      if (playlist.repeat_ === false) {
+
+        // Make sure we stop at the end
+        nextIndex = Math.min(playlist.currentIndex_ + 1, list.length - 1);
+
+      // Repeat
+      } else {
+        nextIndex = playlist.currentIndex_ + 1;
```
Comment (gkatsev) [inline src/playlist-maker.js]:
```js
nextIndex = (plalist.currentIndex_ + 1) % list.length;
```

But really, what you have is just fine.

File: src/playlist-maker.js (hunk 1 above)
Comment (incompl) [inline src/playlist-maker.js]: I'm currently changing it so shuffle is an operation rather than an option.

File: src/playlist-maker.js (hunk 2 above)
Comment (incompl) [inline src/playlist-maker.js]: I usually lean toward the "don't make me think" approach heh.

**Pull request closed without merging**